src/thor/trace_route_action.cc

```diff
@@ -389,12 +390,32 @@ void thor_worker_t::build_route(
     add_path_edge(destination_location, dest_segment->edgeid, dest_segment->target, dest_match.lnglat,
                   dest_match.distance_from);
 
+    // TODO: maybe move this into form path...
+    // build up the discontinuities so we can trim shape where we do uturns
+    route_discontinuities.clear();
```

Comment:
by not filling out the discontinuities (something we've been meaning to do for a while) we are not informing triplegbuilder where to trim the shape of certain PathInfos/Edges. because of this we were getting whole edges even if there was a uturn on the edge. now we fill out the "trimming information" (should probably rename discontinuities to that..) so that the buidler can both cut the shape and set the edge length correctly